streams/tck/src/main/java/org/eclipse/microprofile/reactive/streams/tck/OnErrorResumeStageVerification.java

```diff
@@ -183,10 +187,95 @@ public void onErrorResumeWithPublisherShouldBeAbleToInjectAFailure() {
       .run(getEngine()));
   }
 
-  @Override
-  List<Object> reactiveStreamsTckVerifiers() {
-    return Collections.emptyList();
-  }
+    @Override
+    List<Object> reactiveStreamsTckVerifiers() {
+        return Arrays.asList(
+            new OnErrorResumeWithVerification(),
+            new OnErrorResumeVerification(),
+            new OnErrorResumeWithPublisherVerification()
+        );
+    }
+
+    public class OnErrorResumeWithVerification extends StageProcessorVerification<Integer> {
+
+        @Override
+        public Processor<Integer, Integer> createIdentityProcessor(int bufferSize) {
+            return ReactiveStreams.<Integer>builder()
+                .onErrorResumeWith(ReactiveStreams::failed)
+                .map(Function.identity()).buildRs(getEngine());
+        }
+
+        @Override
+        public Publisher<Integer> createFailedPublisher() {
+            return ReactiveStreams.<Integer>failed(new RuntimeException("failed"))
+                .onErrorResumeWith(t -> {
+                    // Re-throw the exception.
+                    if (t instanceof RuntimeException) {
+                        throw (RuntimeException) t;
+                    }
+                    // Wrap if required.
+                    throw new RuntimeException(t);
+                })
+                .buildRs(getEngine());
+        }
+
+        @Override
+        public Integer createElement(int element) {
+            return element;
+        }
+
+        @Override
+        public void required_spec203_mustNotCallMethodsOnSubscriptionOrPublisherInOnError() {
+            // Ignore this test, as the subscription happen in the onError method to switch the control to the new stream.
```

Comment:
@jroper this is a bit annoying. What do you think?

Reply:
Looks sus - sounds like either rxjava or the `ConnectableProcessor` is doing something wrong here if this is failing. Let me know if you'd like me to take a look.